lib/dockerregistry/inventory.go

```diff
@@ -1434,19 +1439,71 @@ func mkPopulateRequestsForPromotionEdges(
 	}
 }
 
-func (sc *SyncContext) getPromotionEdgesFromManifests(
-	mfests []Manifest) map[PromotionEdge]interface{} {
+// GetPromotionEdgesFromManifests generates all "edges" that we need to promote.
+func (sc *SyncContext) GetPromotionEdgesFromManifests(
+	mfests []Manifest,
+	readRepos bool,
+) map[PromotionEdge]interface{} {
 
 	edges := ToPromotionEdges(mfests)
+
+	if readRepos {
+		regs := getRegistriesToRead(edges)
+		for _, reg := range regs {
+			fmt.Println("reading this reg:", reg)
+		}
+		sc.ReadRegistries(
+			regs,
+			// Do not read these registries recursively, because we already know
+			// exactly which repositories to read (getRegistriesToRead()).
+			false,
+			MkReadRepositoryCmdReal)
+	}
+
 	return sc.getPromotionCandidates(edges)
 }
 
+// getRegistriesToRead collects all unique Docker repositories we want to read
```

Comment:
Should this be called getRepositoriesToRead ?  Admittedly the difference gets me every time.

Reply:
I feel the same --- but I will do a global s/registry/repository in the future (don't want to tackle that here...).